mcstatus/__main__.py

```diff
@@ -1,75 +1,103 @@
 from __future__ import annotations
 
+import sys
 import argparse
 import socket
+from typing import TYPE_CHECKING
 from json import dumps as json_dumps
 
-from mcstatus import JavaServer
+from mcstatus import JavaServer, BedrockServer
+from mcstatus.responses import JavaStatusResponse
 
+if TYPE_CHECKING:
+    SupportedServers = JavaServer | BedrockServer
 
-def ping(server: JavaServer) -> None:
-    print(f"{server.ping()}ms")
+def ping(server: SupportedServers) -> int:
+    print(f"{server.status().latency}")  # XXX: JavaServer.ping() not working?
+    return 0
 
 
-def status(server: JavaServer) -> None:
+def status(server: SupportedServers) -> int:
     response = server.status()
-    if response.players.sample is not None:
-        player_sample = str([f"{player.name} ({player.id})" for player in response.players.sample])
+
+    java_res = response if isinstance(response, JavaStatusResponse) else None
+
+    if not java_res:
+        player_sample = ""
+    elif java_res.players.sample is not None:
+        player_sample = str([f"{player.name} ({player.id})" for player in java_res.players.sample])
     else:
         player_sample = "No players online"
 
-    print(f"version: v{response.version.name} (protocol {response.version.protocol})")
-    print(f'motd: "{response.motd}"')
-    print(f"players: {response.players.online}/{response.players.max} {player_sample}")
+    if player_sample:
+        player_sample = " " + player_sample
+
+    print(f"version: {server.kind()} {response.version.name} (protocol {response.version.protocol})")
+    print(f'motd: {response.motd.to_ansi()}')
+    print(f"players: {response.players.online}/{response.players.max}{player_sample}")
+    print(f"ping: {response.latency:.2f} ms")
+    return 0
 
 
-def json(server: JavaServer) -> None:
+def json(server: SupportedServers) -> int:
     data = {}
     data["online"] = False
+    data["kind"] = server.kind()
     # Build data with responses and quit on exception
     try:
         status_res = server.status(tries=1)
+        java_res = status_res if isinstance(status_res, JavaStatusResponse) else None
         data["version"] = status_res.version.name
         data["protocol"] = status_res.version.protocol
         data["motd"] = status_res.motd.raw
         data["player_count"] = status_res.players.online
         data["player_max"] = status_res.players.max
```

Comment:
I think it's nice to decouple the returned dict from the internal representation, in case the internal representation changes? I don't feel strongly, however.

Reply:
Change of internal representation is a breaking change anyway, so I would rather think about breaking change than maintaining two representations

Reply:
I tried using asdict() and it looks like it will require substantially more internal changes. At the moment, it includes far too much such as entire base64 strings and every field in the special MOTD dataclass. I think this will need more work in another change request.

Reply:
Yeah, for MOTD you have to do special handling

```py
object.__setattr__(status_res, "motd", status_res.motd.minecraft) # the beauty of python
return dataclasses.asdict(status_res)
```

Reply:
That's... rather unpleasant 😅 Better than my plan, though (copying and rewiring addict()'s internals).